app/lib/katello/capsule_content.rb

```diff
@@ -53,6 +53,49 @@ def consumer
       @consumer
     end
 
+    def sync_tasks
+      ForemanTasks::Task.for_resource(self.capsule)
+    end
+
+    def active_sync_tasks
+      sync_tasks.where(:result => 'pending')
+    end
+
+    def last_failed_sync_tasks
+      sync_tasks.where('started_at > ?', last_sync_time).where.not(:result => 'pending')
+    end
+
+    def last_sync_time
+      task = sync_tasks.where.not(:ended_at => nil).where(:result => 'success').order(:ended_at).last
+      task.ended_at unless task.nil?
+    end
+
+    def environment_syncable?(env)
+      last_sync_time.nil? || env.content_view_environments.where('updated_at > ?', last_sync_time).any?
+    end
+
+    def pulp_repositories_data(environment = nil, content_view = nil)
+      @pulp_repositories ||= @capsule.pulp_repositories
+
+      if environment
+        if content_view
+          filter = /#{environment.organization.label}+-#{environment.label}-#{content_view.label}-[^-]+-[^-]+/
+        else
+          filter = /#{environment.organization.label}+-#{environment.label}-[^-]+-[^-]+-[^-]+/
+        end
```

Comment:
it also doesn't tie the format of the label to anything special, as its not tied to that today even if it is generated with a particular structure